### DISCUSSION TITLE
Send data to application from xpra html client

### DISCUSSION THREAD

**Discussion by CH-Evgeny** [Xpra-org/xpra-html5#256] [Q&A]
Hello,
is there any way to send data such as a cookie or a header from xpra-html5 client through xpra server to application that runs on a server?

**Comment by totaam** [comment]
There are many ways this can be achieved.
OTOH, the easiest way might be to use a custom authentication module and have it set a root window property that the application can read.

**Reply by weimerb-tng** [reply]
Hi @totaam,

thanks for your answer, I'm not sure if I understand what you are suggesting though. Is there already a custom authentication module that can be used in our case or do you propose to write a new one?
I'm working together with @CH-Evgeny, let me describe the setup and what we want to achieve we currently have a bit more.
We are using xpra server, that also serves the xpra-html5 client. xpra starts some application with
```
xpra start --start-child-after-connect="application.sh -$PARMS" ...
```
We would like to pass an oidc access token to that application. In front of xpra we have an oauth2-proxy that handles the authentication for our application. The oauth2 proxy forwards the access token to xpra server in a HTTP header (X-Forwarded-Access-Token).
Is there an easy way to forward the HTTP header that the xpra server receives to the application we start with xpra?

**Comment by totaam** [comment]
_custom_ implies writing a new one, though it can be based on existing ones.

How you send the token to the application can be done via X11 root window properties or any other RPC mechanism you choose to use. (ie: `dbus`, touching a file, whatever is easier to implement)
